runtime/src/current_sandbox.c

```diff
@@ -88,20 +103,43 @@ current_sandbox_start(void)
 	current_sandbox_disable_preemption(sandbox);
 	sandbox->completion_timestamp = __getcycles();
 
-	/* Retrieve the result, construct the HTTP response, and send to client */
-	if (sandbox_send_response(sandbox) < 0) {
-		error_message = "Unable to build and send client response\n";
-		goto err;
-	};
 
-	http_total_increment_2xx();
+	if (next_module != NULL) {
+		/* Generate a new request, copy the current sandbox's output to the next request's buffer, and put it to the global queue */
+		ssize_t output_length = sandbox->request_response_data_length - sandbox->request_length;
+		char * pre_func_output = (char *) malloc(output_length);
+		memcpy(pre_func_output, sandbox->request_response_data + sandbox->request_length, output_length);
```

Comment:
> I don't follow the logic here with the `+ request_length`. Does the sledge code (before your addition) use the same buffer for holding the request _and_ the response? Sorry for forgetting here!

Yes, this is also my confusion when I saw sledge code at the first time. This is the original code design. It use the same buffer for holding the request and response. See the name `request_response_data`.  Also see the comment of the function to prepare http response.
```
static inline int
sandbox_send_response(struct sandbox *sandbox)
{
        assert(sandbox != NULL);

        /*
         * At this point the HTTP Request has filled the buffer up to request_length, after which
         * the STDOUT of the sandbox has been appended. We assume that our HTTP Response header is
         * smaller than the HTTP Request header, which allows us to use memmove once without copying
         * to an intermediate buffer.
         */
        memset(sandbox->request_response_data, 0, sandbox->request_length);
```

Reply:
@lyuxiaosu  is right, the Sledge code does use the same buffer for both the request and the response. My only comment about the above code, I don't see why creating an extra variable for the output is really needed (+malloc). The `responspe_data `with the proper output cursor would be sufficient.

Reply:
I cannot find `responspe_data`, are you mentioning `sandbox->request_response_data`? The malloc here is for copying the currency sandbox output to the next sandbox's request buffer, when the next sandbox running, it won't read the input from the networking if it is the middle sandbox in the chain, it will read the input from this malloced buffer, otherwise, when the current sandbox is freed, its `request_response_data` is freed.